python/cudf/tests/test_groupby.py

```diff
@@ -35,16 +37,52 @@ def get_nelem():
         yield elem
 
 
+@pytest.fixture
+def gdf():
+    return DataFrame({'x': [1, 2], 'y': [1, 2]})
+
+
+@pytest.fixture
+def pdf(gdf):
+    return gdf.to_pandas()
+
+
+def test_groupby_no_index(pdf, gdf):
+    gdf = gdf.groupby('y', as_index=False).agg({'x': 'mean'})
+    pdf = pdf.groupby('y', as_index=False).agg({'x': 'mean'})
+    assert_eq(pdf, gdf)
+
+
+def test_groupby_index(pdf, gdf):
+    gdf = gdf.groupby('y', as_index=True).agg({'x': 'mean'})
+    pdf = pdf.groupby('y', as_index=True).agg({'x': 'mean'})
+    assert_eq(pdf, gdf)
```

Comment:
You might consider using pytest parameters here

```python
@pytest.mark.parametrize('as_index', [True, False])
def test_groupby_index(pdf, gdf, as_index):
    gdf = gdf.groupby('y', as_index=as_index).agg({'x': 'mean'})
    pdf = pdf.groupby('y', as_index=as_index).agg({'x': 'mean'})
    assert_eq(pdf, gdf)
```

This would reduce code duplication a bit and also help us if we need to refactor these tests again in the future (from your commit messages it looks like you might have run into some frustration there with the previous tests).